content/vendor/guides/using-ecr-private-images.md

```diff
@@ -0,0 +1,319 @@
+To use `docker login` with ECR, we will need to configure AWS CLI with the AWS Access Key ID and AWS Secret Key for this user.
+
+Let's start by pulling the public image
+
+```shell
+$ docker pull nginx
+```
+
+You should have an output similar to this:
+
+```shell
+Using default tag: latest
+latest: Pulling from library/nginx
+d121f8d1c412: Pull complete 
+ebd81fc8c071: Pull complete 
+655316c160af: Pull complete 
+d15953c0e0f8: Pull complete 
+2ee525c5c3cc: Pull complete 
+Digest: sha256:c628b67d21744fce822d22fdcc0389f6bd763daac23a6b77147d0712ea7102d0
+Status: Downloaded newer image for nginx:latest
+docker.io/library/nginx:latest
+```
+
+Next we need to login to ECR and push this container. 
+To use `docker login` with ECR we will need to [install the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) and  [configure it](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html) if not already.
+As part of this, we will need to provide the AWS Access Key ID and AWS Secret Key for a user that has permissions to create and push images to the repository. Please refer to [this guide](https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-cli.html) if you are not familiar using the AWS CLI to work with containers and ECR.
+
+
+Just like with any other private registry, we need to know the registry endpoint to pass the `docker login` command.
+The syntax is as follows:
+
+```shell
+
+docker login [some.private.registry]:[port]
+
+```
+In this case, the endpoint is the **[some.private.registry]:[port]**
+
+To determine the endpoint for ECR, login to the AWS console and seach for 'ECR', which should bring up Elastic Container Registry as an option as shown below.
+
+![search-4-ecr](/images/guides/kots/priv-reg-ecr-search-4-ecr.png)
+
+Select 'Elastic Container Registry' from the options in the dropdown to get to the list of repositories.
+
+![ecr-repos](/images/guides/kots/priv-reg-ecr-repos.png)
+
+As you can see from the screenshot above, you can see the endpoints for each repository under the URI column.
+For the purpose of this guide, we will push the nginx image to the **demo-apps** repository.
+ 
+To determine the endpoint to use in the login command, use the url without the repository name.
+
+When it comes to loggin in to ECR, we need to use the AWS CLI to the user credentials.
+As an example, to login to ECR we run the following command:
+
+```shell
+
+$ aws ecr get-login-password --region us-east-2 | docker login --username AWS --password-stdin 4999999999999.dkr.ecr.us-east-2.amazonaws.com
+```
+
+A succesful login will display a `Login Succeeded` message.
+In order to push this image to our private repsoitory, we need to tag the image.
+The new tag will consist of:
+
+`<ecr repoendpoint>/image`
+
+As an example, to tag the public nginx image we run the following command:
+
+```shell
+$ docker tag nginx 4999999999999.dkr.ecr.us-east-2.amazonaws.com/demo-apps/nginx
+```
```

Comment:
My only comment here is that for the push to work using `demo-apps/nginx` the repository must be created with the name `demo-apps/nginx` which might trip up some users (like me) that aren't used to the AWS flavor of repository.